wandb/util.py

```diff
@@ -810,6 +812,19 @@ def default(self, obj: Any) -> Any:
         return json.JSONEncoder.default(self, obj)
 
 
+class WandBJSONEncoderWeave(json.JSONEncoder):
+    """A JSON Encoder for weave objects, used by StreamTable today."""
+
+    def default(self, obj: Any) -> Any:
+        obj, converted = json_friendly(obj)
```

Comment:
i still feel like `json_friendly` has too much business logic in it. I'd prefer not to bring along all this legacy logic and just say that if something is a non-primitive (list, dict, none, float, int, string, bool) then replace with None.

Reply:
`json_friendly` handles `numpy` arrays, `tensors`, `fp16`, `fp128`, etc.  Seemed like a good things to handle...